### PR TITLE
Fix k8s postgres quote passthrough

### PR DESCRIPTION
**Pull request by jon-funk**
- Fixes quote passthrough

Note: opened for CR but I'm going to test on a few queries in the meantime

### CODE REVIEW

**Comment by jon-funk** [comment]
Having the query in the /tmp file was super helpful for troubleshooting btw - let me quickly identify how the query was getting parsed and sent in.

**Comment by jon-funk** [comment]
`"` are getting dropped at the tail end of the passthrough, investigating..

**Comment by jon-funk** [comment]
Tested for both the runbook and sli. Let me know if you hit any issues.

**Comment by stewartshea** [comment]
I just tried a single quote around the string: 

- From the CLI
```
# SELECT pid,now() - pg_stat_activity.query_start AS duration,query,state FROM pg_stat_activity WHERE (now() - pg_stat_activity.query_start) > interval '5 minutes' AND state = 'active';
 pid |        duration        |                   query                    | state  
-----+------------------------+--------------------------------------------+--------
 402 | 2 days 01:29:58.224487 | START_REPLICATION 1A7/46000000 TIMELINE 14 | active

```

robot ouput shows the command as: 
```
SELECT pid,now() - pg_stat_activity.query_start AS duration,query,state FROM pg_stat_activity WHERE (now() - pg_stat_activity.query_start) > interval 5 minutes AND state = active;
```

Double quotes don't work with this query, and if attempted, fails.

**Comment by stewartshea** [comment]
Btw, this is related to internal issue https://github.com/runwhen/platform-core/issues/739 for ref.

File: libraries/RW/Postgres/postgres.py [hunk 1]
```diff
@@ -79,6 +75,19 @@ def parse_metric_and_time(
...
+    def quote_query(query: str) -> str:
+        """Simple helper method to escape specific characters that cause issues in the pod exec passthrough
+
+        Args:
+            query (str): _description_
+
+        Returns:
+            str: _description_
+        """
+        query = query.replace('"', '\\\\\\"')
```
Comment (stewartshea) [inline libraries/RW/Postgres/postgres.py:92]: the problem is with single quotes; maybe doubles are also an issue, but related to the queries we were attempting, we need single quotes escaped.

**Review by stewartshea — CHANGES_REQUESTED** [review]

**Comment by stewartshea** [comment]
I was testing this on the wrong codebundle. so at first was chasing down a double/single quote issue but this seems to work in the codebundles you have edited here. I'm going to add in the database triage to see if it works too.

**Comment by jon-funk** [comment]
ack - thought I was losing my mind lol - let me know if you hit anything there after making the changes

**Comment by stewartshea** [comment]
Hah, I thought I was losing my mind too! 

This looks like it works just fine! Nice work on this! 
image

**Pull request merged**